### ISSUE TITLE
Add new images

### ISSUE THREAD

**Issue by szabgab** [kantord/LibreLingo#1859]
I wonder how should the requesting and adding of new images be handled. In any case I'd like to ask for the following images to be added:

```
apple
water
twenty
flower
fruit
plum
watermelon
basketball
honey
party
```

**Comment by kantord** [comment]
> I wonder how should the requesting and adding of new images be handled.

I think that it should be handled in a similar way as audio files are handled. Maybe even the audio and image fetcher logic should be merged into one 'external resource' managed/fetcher.

My idea is that in your yaml files, you could directly link pictures from https://unsplash.com/ that would in turn be automatically fetched and added to metadata files in order to be able to keep track of where pictures come from and what their license is etc. 

This way there will be no limitations as to what images a course can use and even the handling of image formats can be automated as well. The comression

**Comment by szabgab** [comment]
Is there an example how to link images from https://unsplash.com/ in the yaml files?

**Comment by kantord** [comment]
> Is there an example how to link images from https://unsplash.com/ in the yaml files?

keep in mind I'm talking hypothetically, this feature is not implemented yet.

but I'm thinking to do the simplest possible way and just copy the link of the page where you can see the image on unsplash: 

```yaml
https://unsplash.com/photos/VgIG-PKmgxg
```

**Comment by kantord** [comment]
I think perhaps, if your course is very experimental for now, you can start putting this value already?

the idea is that you put the image name is it exists, if it doesn't you put a link. I think you won't get any errors, just those images won't work. But I think it should not cause any issues with any other aspect of working on the course. So if you only have a few images like this, it should not be a big deal at least until the course is released to the public.

Then once the course is in 'beta' status then we can either have this already developed, or if this feature is not ready yet, it will be easy to instead add the new images the old fashioned way, because we can use the yaml files themselves in order to find a list of images we need to add

```bash
grep -r . https://unsplash.com/photos
```

**Comment by szabgab** [comment]
For now I just added names of images that don't exist and I have a script in https://github.com/szabgab/LibreLingo-tools that reports all the image names that are used but don't exist in our repo.

I could start adding links to unsplash photos, but they will need to resized before displaying them, don't they?
And what if some images are removed from unsplash? How would that be handled?

**Comment by kantord** [comment]
> And what if some images are removed from unsplash? How would that be handled?

I think the images have to be kept in the repo. We also need to keep a metadata file with data about all the pictures. What was the original URL author etc etc. So if an image is removed we still keep it and we keep the metadata. What do you think?

**Comment by kantord** [comment]
> but they will need to resized before displaying them, don't they?

yes. right now there's some rudimentary tooling for this here: https://github.com/kantord/LibreLingo/blob/main/apps/web/scripts/fetchPhoto.sh

Long term we probably need something more sophisticated.
